### DISCUSSION TITLE
friendly_name '0xxxxx' is already in use during renaming device

### DISCUSSION THREAD

**Discussion by alp-by** [Koenkk/zigbee2mqtt#23341] [General]
Hi.
I added Zigbee device with a name '0x00124b002fa6ac15'. Deviced was added successfully. Later on I decided to change (updating HA devices) the name  of the device to "0x00124b002fa6ac15_blue1". Than worked without issues.
![image](https://github.com/user-attachments/assets/4668ffe9-9cc4-452a-9767-85a396edc321)

Later on I decided to change the device name back to "0x00124b002fa6ac15" and got the error: 
![image](https://github.com/user-attachments/assets/29ca5821-5e96-41dd-91fd-4cd4d7d5a6cd)

I checked the devices list in HA, I didn't find any devices with "0x00124b002fa6ac15". That's the only what I have:
![image](https://github.com/user-attachments/assets/0e92ecc5-5973-4997-95e0-ff23e721dccb)

The same info with entities. (no entities with 0x00124b002fa6ac15*, only 0x00124b002fa6ac15_blue1*

I use the latest z2m version 1.39.0-1 and latest HA core.
Is it a bug of may be I do smth wrong?

**Comment by dagobert** [comment]
I am not sure, if this is still a problem for you...

I had the same problem, summerized in the following sentence:

Change name of a zigbee2mqqt device from the original `0xabcde12345` name to eg. `My-New-Device-Name`, than change it back to the original name `0xabcde12345`.

The problem seems to exist within zigbee2mqtt. I research all HomeAssistant files but there were not leftovers. The only place that hinted to the problem was the zigbee2mqtt configuration.yaml

To solve the problem you need to do the following:

1. Stop zigbee2mqtt
2. Open the file zigbee2mqtt/configuration.yaml
3. Look for the section `devices:`
4. Within this section you will find en entry with your original device name `0xabcde12345` and within this section the entry `friendly_name: My-New-Device-Name`
5. Change the entry `friendly_name: My-New-Device-Name` to `friendly_name: '0xabcde12345'`
6. Start zigbee2mqtt

After zigbee2mqtt is running again, you can check the result.

Example from:
```
devices:
  '0xabcde12345':
    friendly_name: My-New-Device-Name
```

To:
```
devices:
  '0xabcde12345':
    friendly_name: '0xabcde12345'
```
